### PR TITLE
docs - update filter docs

### PR DESCRIPTION
**Pull request by jeff-bruemmer**
Clarifies filter options when editing table metadata.

Re: limiting search results in filters, looks like the limit is set here: https://github.com/metabase/metabase/blob/a9e0b014edeee0a4cfb00ab4f36f1b49a8023f28/src/metabase/api/field.clj#L39

### CODE REVIEW

File: docs/data-modeling/metadata-editing.md [hunk 1]
```diff
@@ -144,14 +144,13 @@ To change a column's [filter widget](../dashboards/filters.md):
...
-- **Plain input box**: Enter a search term and Metabase will make autocomplete suggestions for the search (no checkboxes).
+The default behavior for the `Is` filter for the field.
 
-#### Default filters
+- **Search box**: Display a search box and suggest autocompletions for values in that column that match the search term(s).
+- **A list of all values**: Display a search box, as well as a list of checkboxes for values in a dropdown menu that people can select as search terms.
+- **Plain input box**: Display a search box, but do NOT suggest autocompletions.
 
-- Columns with more than 100 unique values will default to a plain input box filter.
-- Columns with fewer values will display a search box filter.
```
Comment (nemanjaglumac) [inline docs/data-modeling/metadata-editing.md:154]: Why are we removing this part?

File: docs/data-modeling/metadata-editing.md [hunk 2]
```diff
@@ -144,14 +144,13 @@ To change a column's [filter widget](../dashboards/filters.md):
...
+For the `Is` filter option, Metabase will accept multiple search terms. The `Is` filter will return results that match any of the terms (e.g., "term 1" OR "term 2").
```
Comment (nemanjaglumac) [inline docs/data-modeling/metadata-editing.md:153]: This is now changing with the multi-select filter options.
We're introducing this functionality to all string filter options, except for "is empty" / "is not empty".

File: docs/data-modeling/metadata-editing.md [hunk 3]
```diff
@@ -144,14 +144,13 @@ To change a column's [filter widget](../dashboards/filters.md):
...
 
-- **Search box**: Enter a search term and Metabase will display checkboxes for column values that match the search.
-- **A list of all values**: dropdown menu with checkboxes for all column values.
-- **Plain input box**: Enter a search term and Metabase will make autocomplete suggestions for the search (no checkboxes).
+The default behavior for the `Is` filter for the field.
 
-#### Default filters
+- **Search box**: Display a search box and suggest autocompletions for values in that column that match the search term(s).
+- **A list of all values**: Display a search box, as well as a list of checkboxes for values in a dropdown menu that people can select as search terms.
+- **Plain input box**: Display a search box, but do NOT suggest autocompletions.
```
Comment (nemanjaglumac) [inline docs/data-modeling/metadata-editing.md:151]: This part is much more clear. Thanks.
Maybe this should be the whole diff for now?

File: docs/data-modeling/metadata-editing.md (hunk 1 above)
Comment (jeff-bruemmer) [reply docs/data-modeling/metadata-editing.md:154]: The limit is no longer 100. Can you confirm that the limit is 1000?

File: docs/data-modeling/metadata-editing.md (hunk 2 above)
Comment (jeff-bruemmer) [reply docs/data-modeling/metadata-editing.md:153]: Removed.

File: docs/data-modeling/metadata-editing.md (hunk 1 above)
Comment (nemanjaglumac) [reply docs/data-modeling/metadata-editing.md:154]: I'll try to find it.

What about the second line that claimed that "Columns with fewer values will display a search box filter."?

Comment (jeff-bruemmer) [reply docs/data-modeling/metadata-editing.md:154]: That line also depends on the limit.

File: docs/data-modeling/metadata-editing.md (hunk 3 above)
Comment (jeff-bruemmer) [reply docs/data-modeling/metadata-editing.md:151]: Updated. We can also ship this, and add a note about the limit in a subsequent PR.

**Review by nemanjaglumac — APPROVED** [review]

**Pull request merged**